pkg/irreparabledb/irreparabledb.go

```diff
@@ -0,0 +1,196 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package irreparabledb
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/internal/migrate"
+	"storj.io/storj/pkg/auth"
+	dbx "storj.io/storj/pkg/irreparabledb/dbx"
+	"storj.io/storj/pkg/pb"
+	pdbauth "storj.io/storj/pkg/pointerdb/auth"
+)
+
+var (
+	mon = monkit.Package()
+)
+
+// Server implements the statdb RPC service
+type Server struct {
+	DB     *dbx.DB
+	logger *zap.Logger
+}
+
+// NewServer creates instance of Server
+func NewServer(driver, source string, logger *zap.Logger) (*Server, error) {
+	db, err := dbx.Open(driver, source)
+	if err != nil {
+		return nil, err
+	}
+
+	err = migrate.Create("irreparabledb", db)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Server{
+		DB:     db,
+		logger: logger,
+	}, nil
+}
+
+func (s *Server) validateAuth(ctx context.Context) error {
+	APIKeyBytes, ok := auth.GetAPIKey(ctx)
+	if !ok {
+		return Error.New("no api key was provided")
+	}
+	if !pdbauth.ValidateAPIKey(string(APIKeyBytes)) {
```

Comment:
Can you combine both `if` statements into one? If no APIKey was provided then `"Invalid API credential"` error is also good.

Reply:
Done